### ISSUE TITLE
CursorIndexOutOfBoundsException

### ISSUE THREAD

**Issue by Lir10** [coomar2841/android-multipicker-library#121]
**Vivo 1713 
Android 7.0**

```
 Exception java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=8666, result=-1, data=Intent { dat=content://com.android.contacts/contacts/lookup/2081i6018604b0ad7d62f.2012r21132-2F514B352914/21231?encrypt= < 2 flg=0x1 }} to activity {com.test.test/com.test.test.CombineScreen}: android.database.CursorIndexOutOfBoundsException: Index 0 requested, with a size of 0
android.app.ActivityThread.deliverResults (ActivityThread.java:4245)
android.app.ActivityThread.handleSendResult (ActivityThread.java:4288)
android.app.ActivityThread.-wrap20 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1630)
android.os.Handler.dispatchMessage (Handler.java:110)
android.os.Looper.loop (Looper.java:203)
android.app.ActivityThread.main (ActivityThread.java:6363)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1076)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:937)
arrow_drop_down
Caused by android.database.CursorIndexOutOfBoundsException: Index 0 requested, with a size of 0
android.database.AbstractCursor.checkPosition (AbstractCursor.java:460)
android.database.AbstractWindowedCursor.checkPosition (AbstractWindowedCursor.java:151)
android.database.AbstractWindowedCursor.getInt (AbstractWindowedCursor.java:83)
android.database.CursorWrapper.getInt (CursorWrapper.java:122)
com.kbeanie.multipicker.api.ContactPicker.getRawContactId (ContactPicker.java:123)
com.kbeanie.multipicker.api.ContactPicker.queryForContact (ContactPicker.java:148)
com.kbeanie.multipicker.api.ContactPicker.submit (ContactPicker.java:110)
com.test.test.CombineScreen.onActivityResult (CombineScreen.java:133)
android.app.Activity.dispatchActivityResult (Activity.java:6944)
android.app.ActivityThread.deliverResults (ActivityThread.java:4241)
android.app.ActivityThread.handleSendResult (ActivityThread.java:4288)
android.app.ActivityThread.-wrap20 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1630)
android.os.Handler.dispatchMessage (Handler.java:110)
android.os.Looper.loop (Looper.java:203)
android.app.ActivityThread.main (ActivityThread.java:6363)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1076)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:937)

```

**Comment by coomar2841** [comment]
Does this happen for all contacts? Or only a few?

**Comment by Lir10** [comment]
@coomar2841 looks like it's for all of them

I have more crashes like that from different devices: 

**Device: vivo Y21L 
API Level 22**
```
 Exception java.lang.RuntimeException: Unable to resume activity {com.test.test/com.test.test.CombineScreen}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=8666, result=-1, data=Intent { dat=content://com.android.contacts/contacts/lookup/3176r37771-4D4541414F3739/37810?encrypt= < 2 flg=0x1 }} to activity {com.test.test/com.test.test.CombineScreen}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.kbeanie.multipicker.api.ContactPicker.submit(android.content.Intent)' on a null object reference
android.app.ActivityThread.performResumeActivity (ActivityThread.java:3122)
android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3153)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2514)
android.app.ActivityThread.access$900 (ActivityThread.java:175)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1382)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:135)
android.app.ActivityThread.main (ActivityThread.java:5425)
java.lang.reflect.Method.invoke (Method.java)
java.lang.reflect.Method.invoke (Method.java:372)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:928)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:723)

Caused by java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=8666, result=-1, data=Intent { dat=content://com.android.contacts/contacts/lookup/3176r37771-4D4541414F3739/37810?encrypt= < 2 flg=0x1 }} to activity {com.mynavy.onetoonelocator/com.mynavy.onetoonelocator.CombineScreen}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.kbeanie.multipicker.api.ContactPicker.submit(android.content.Intent)' on a null object reference
android.app.ActivityThread.deliverResults (ActivityThread.java:3710)
android.app.ActivityThread.performResumeActivity (ActivityThread.java:3094)
android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3153)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2514)
android.app.ActivityThread.access$900 (ActivityThread.java:175)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1382)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:135)
android.app.ActivityThread.main (ActivityThread.java:5425)
java.lang.reflect.Method.invoke (Method.java)
java.lang.reflect.Method.invoke (Method.java:372)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:928)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:723)

Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'void com.kbeanie.multipicker.api.ContactPicker.submit(android.content.Intent)' on a null object reference
com.test.test.CombineScreen.onActivityResult (CombineScreen.java:133)
android.app.Activity.dispatchActivityResult (Activity.java:6276)
android.app.ActivityThread.deliverResults (ActivityThread.java:3706)
android.app.ActivityThread.performResumeActivity (ActivityThread.java:3094)
android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3153)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2514)
android.app.ActivityThread.access$900 (ActivityThread.java:175)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1382)
android.os.Handler.dispatchMessage (Handler.java:102)
android.os.Looper.loop (Looper.java:135)
android.app.ActivityThread.main (ActivityThread.java:5425)
java.lang.reflect.Method.invoke (Method.java)
java.lang.reflect.Method.invoke (Method.java:372)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:928)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:723)
```

**Device: CPH1717
API Level 25**

 ```
Exception java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=8666, result=-1, data=Intent { dat=content://com.android.contacts/contacts/lookup/3056r602-A8B4D0CCB4C0BE/1865 flg=0x1 }} to activity {com.test.test/com.test.test.CombineScreen}: android.database.CursorIndexOutOfBoundsException: After last row.
android.app.ActivityThread.deliverResults (ActivityThread.java:4458)
android.app.ActivityThread.handleSendResult (ActivityThread.java:4501)
android.app.ActivityThread.-wrap20 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1740)
android.os.Handler.dispatchMessage (Handler.java:110)
android.os.Looper.loop (Looper.java:232)
android.app.ActivityThread.main (ActivityThread.java:6661)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1106)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:967)

Caused by android.database.CursorIndexOutOfBoundsException: After last row.
android.database.MatrixCursor.get (MatrixCursor.java:73)
android.database.MatrixCursor.getInt (MatrixCursor.java:269)
com.kbeanie.multipicker.api.ContactPicker.getRawContactId (ContactPicker.java:123)
com.kbeanie.multipicker.api.ContactPicker.queryForContact (ContactPicker.java:148)
com.kbeanie.multipicker.api.ContactPicker.submit (ContactPicker.java:110)
com.test.test.CombineScreen.onActivityResult (CombineScreen.java:133)
android.app.Activity.dispatchActivityResult (Activity.java:7166)
android.app.ActivityThread.deliverResults (ActivityThread.java:4454)
android.app.ActivityThread.handleSendResult (ActivityThread.java:4501)
android.app.ActivityThread.-wrap20 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1740)
android.os.Handler.dispatchMessage (Handler.java:110)
android.os.Looper.loop (Looper.java:232)
android.app.ActivityThread.main (ActivityThread.java:6661)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1106)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:967)
```

**Device: CPH1717
API Level 25**
```
Exception java.lang.RuntimeException: Unable to resume activity {com.test.test/com.test.test.CombineScreen}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=8666, result=-1, data=Intent { dat=content://com.android.contacts/contacts/lookup/3056r602-A8B4D0CCB4C0BE/1865 flg=0x1 }} to activity {com.test.test/com.test.test.CombineScreen}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.kbeanie.multipicker.api.ContactPicker.submit(android.content.Intent)' on a null object reference
android.app.ActivityThread.performResumeActivity (ActivityThread.java:3725)
android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3775)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2968)
android.app.ActivityThread.handleRelaunchActivity (ActivityThread.java:4906)
android.app.ActivityThread.-wrap19 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1690)
android.os.Handler.dispatchMessage (Handler.java:110)
android.os.Looper.loop (Looper.java:232)
android.app.ActivityThread.main (ActivityThread.java:6661)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1106)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:967)

Caused by java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=8666, result=-1, data=Intent { dat=content://com.android.contacts/contacts/lookup/3056r602-A8B4D0CCB4C0BE/1865 flg=0x1 }} to activity {com.test.test/com.test.test.CombineScreen}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.kbeanie.multipicker.api.ContactPicker.submit(android.content.Intent)' on a null object reference
android.app.ActivityThread.deliverResults (ActivityThread.java:4458)
android.app.ActivityThread.performResumeActivity (ActivityThread.java:3693)
android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3775)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2968)
android.app.ActivityThread.handleRelaunchActivity (ActivityThread.java:4906)
android.app.ActivityThread.-wrap19 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1690)
android.os.Handler.dispatchMessage (Handler.java:110)
android.os.Looper.loop (Looper.java:232)
android.app.ActivityThread.main (ActivityThread.java:6661)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1106)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:967)

Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'void com.kbeanie.multipicker.api.ContactPicker.submit(android.content.Intent)' on a null object reference
com.test.test.CombineScreen.onActivityResult (CombineScreen.java:133)
android.app.Activity.dispatchActivityResult (Activity.java:7166)
android.app.ActivityThread.deliverResults (ActivityThread.java:4454)
android.app.ActivityThread.performResumeActivity (ActivityThread.java:3693)
android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3775)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2968)
android.app.ActivityThread.handleRelaunchActivity (ActivityThread.java:4906)
android.app.ActivityThread.-wrap19 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1690)
android.os.Handler.dispatchMessage (Handler.java:110)
android.os.Looper.loop (Looper.java:232)
android.app.ActivityThread.main (ActivityThread.java:6661)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1106)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:967)
```

**Comment by coomar2841** [comment]
```
xception: Attempt to invoke virtual method 'void com.kbeanie.multipicker.api.ContactPicker.submit(android.content.Intent)' on a null object reference
```
These crashes mean that ContactPicker instance has been killed and you have to re-initialize them.

For the main issue, I am still not sure.

**Comment by Lir10** [comment]
@coomar2841 is it possible to add any exception handler from the library ?

**Comment by coomar2841** [comment]
Fixed. Will be available with version 1.5.

**Comment by coomar2841** [comment]
Released with v1.5
